src/main/java/io/github/thebusybiscuit/slimefun4/utils/SlimefunUtils.java

```diff
@@ -302,10 +324,20 @@ public static boolean isItemSimilar(@Nullable ItemStack item, @Nullable ItemStac
                  */
                 ItemMeta possibleSfItemMeta = sfitem.getItemMeta();
                 Debug.log(TestCase.CARGO_INPUT_TESTING, "  sfitem is ItemStackWrapper - possible SF Item: {}", sfitem);
+                String id = Slimefun.getItemDataService().getItemData(itemMeta).orElse(null);
 
                 // Prioritize SlimefunItem id comparison over ItemMeta comparison
-                if (Slimefun.getItemDataService().hasEqualItemData(possibleSfItemMeta, itemMeta)) {
+                if (Slimefun.getItemDataService().getItemData(possibleSfItemMeta).orElse("").equals(id)) {
```

Comment:
Move the getItemData().orElse out of the if. Then also make the `orElse` be null, I don't like making a new string here.